### PR TITLE
[clang] Parse attribute [[gnu::no_stack_protector]]

### LINKED ISSUE #75235: [clang]  `[[gnu::no_stack_protector]]` is not recognized
According to [GCC trunk](https://godbolt.org/z/3vcq3MrK1), `[[gnu::no_stack_protector]]` is a valid attribute to suppress stack protector, however, `clang` does not respect this attribute.

https://github.com/llvm/llvm-project/blob/31b72b0742b09c970bea47b70eb535fe964e3c8f/clang/include/clang/Basic/Attr.td#L2215

### PR DESCRIPTION
**Pull request by Lancern**
This commit adds relative TableGen definitions to parse the `[[gnu::no_stack_protector]]` attribute.

This PR addresses issue #75235.

### CODE REVIEW

**Review by inclyc — CHANGES_REQUESTED** [review]
The PR contains functional change so please add test & assertions under `clang/test`

**Review by inclyc — APPROVED** [review]

File: clang/test/Parser/gnu-attributes.c [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -std=c23 %s
```
[the remark this answers is not in the document]
Comment (Lancern) [reply clang/test/Parser/gnu-attributes.c:1]: My concern is that names such as `attr-no-stack-protector.cpp` are too "specific". No other single attribute has its dedicated test file under the test directory.

[the remark this answers is not in the document]
Comment (inclyc) [reply clang/test/Parser/gnu-attributes.c:1]: I just noticed that we already have test file at `clang/test/Sema/no_stack_protector.c`, may be just update that file.

> No other single attribute has its dedicated test file under the test directory.

No, just see `no_stack_protector.c`.

[the remark this answers is not in the document]
Comment (inclyc) [reply clang/test/Parser/gnu-attributes.c:1]: PS. for cxx attribute you may create a new file.

**Review by inclyc — APPROVED** [review]

**Pull request merged**